pkg/crc/cleanup/cleanup_linux.go

```diff
@@ -0,0 +1,69 @@
+package cleanup
+
+import (
+	"fmt"
+	"github.com/code-ready/crc/pkg/crc/constants"
+	"os"
+	"strings"
+
+	"github.com/code-ready/crc/pkg/crc/errors"
+	"github.com/code-ready/crc/pkg/crc/machine/libvirt"
+	"github.com/code-ready/crc/pkg/crc/preflight"
+	crcos "github.com/code-ready/crc/pkg/os"
+)
+
+func CleanupHost() error {
+	err := errors.MultiError{}
+	// Remove crc network from libvirt
+	err.Collect(removeCRCLibvirtNetwork())
+
+	// Removing NetworkManager dnsmasq config for crc
+	err.Collect(removeNetworkManagerConfigForDNSMasq())
+
+	// Removing dnsmasq config for crc
+	err.Collect(removeDnsmasqConfigForCRC())
+
+	// Remove the crc directory
+	err.Collect(removeCrcBaseDir())
+
+	return err.ToError()
+}
+
+func removeCRCLibvirtNetwork() error {
+	stdOut, stdErr, err := crcos.RunWithDefaultLocale("virsh", "--connect", "qemu:///system", "net-list")
+	if err != nil {
+		return fmt.Errorf("%+v: %s", err, stdErr)
+	}
+	if !strings.Contains(stdOut, "crc") {
+		return nil
+	}
+	_, stdErr, err = crcos.RunWithDefaultLocale("virsh", "--connect", "qemu:///system", "net-destroy", libvirt.DefaultNetwork)
+	if err != nil {
+		return fmt.Errorf("%+v: %s", err, stdErr)
+	}
+	_, stdErr, err = crcos.RunWithDefaultLocale("virsh", "--connect", "qemu:///system", "net-undefine", libvirt.DefaultNetwork)
+	if err != nil {
+		return fmt.Errorf("%+v: %s", err, stdErr)
+	}
+	return nil
+}
+
+func removeNetworkManagerConfigForDNSMasq() error {
```

Comment:
I think it would be good to keep naming consistent with the preflight checks, so `cleanupCrcNetworkManagerConfig`, `cleanupCrcDnsmasqConfigFile` and so on. Actually, I'd just add a 'cleanup' field to the PreflightCheck structure in https://github.com/cfergeau/crc/blob/preflight-refactor/pkg/crc/preflight/preflight.go , the cleanup command can be generic and only need to iterate over these.